agent/consul/catalog_endpoint.go

```diff
@@ -257,19 +257,21 @@ func (c *Catalog) ListNodes(args *structs.DCSpecificRequest, reply *structs.Inde
 		&args.QueryOptions,
 		&reply.QueryMeta,
 		func(ws memdb.WatchSet, state *state.Store) error {
-			var index uint64
-			var nodes structs.Nodes
 			var err error
 			if len(args.NodeMetaFilters) > 0 {
-				index, nodes, err = state.NodesByMeta(ws, args.NodeMetaFilters)
+				reply.Index, reply.Nodes, err = state.NodesByMeta(ws, args.NodeMetaFilters)
 			} else {
-				index, nodes, err = state.Nodes(ws)
+				reply.Index, reply.Nodes, err = state.Nodes(ws)
```

Comment:
https://github.com/golang/go/blob/go1.14.4/src/net/rpc/server.go#L347-L350

The RPC server takes care of sending an empty response if there are any errors, so the local variables do not appear to be necessary.